recipes/openssl/1.x.x/conanfile.py

```diff
@@ -172,6 +174,14 @@ def config_options(self):
         else:
             del self.options.fPIC
 
+        if self.settings.os == "Emscripten":
+            self.options.no_asm = True
+            self.options.no_deprecated = True
+            self.options.no_dso = True
+            self.options.no_threads = True
+            self.options.no_stdio = True
+            self.options.no_tests = True
```

Comment:
This makes it still possible to run:
```
conan create . openssl/1.1.1k@ -pr emsdk_profile -o openssl:no_stdio=False
```
Should this also be tested in a `validate` method?
```python
def validate(self):
    if self.settings.or == "Emscripten":
        if not all((self.options.no_asm, self.options.no_deprecated, self.options.no_dso, self.options.no_threads, self.options.no_stdio, self.options.no_tests)):
            raise ConanInvalidConfiguration("os=Emscripten requires {no_asm,no_deprecated,no_dso,no_threads,no_stdio,no_tests}=True")
```

Reply:
thanks, I will add this check

Reply:
I'm wondering whether the test should be `self.settings.os == "Emscripten"` or `self.settings.arch == "wasm"`.

Reply:
Does my suggestion actually work? Does it block `-o openssl:no_asm=False`? I didn't verify locally.

Reply:
this is good question! But idk any other sdks that are capable of compiling that complicated code to wasm.

Reply:
> Does my suggestion actually work? Does it block `-o openssl:no_asm=False`? I didn't verify locally.

yes it does